### DISCUSSION TITLE
Our closing logic doesn't work on shutdown event

### DISCUSSION THREAD

**Discussion by Rybenka** [micronaut-projects/micronaut-core#6354] [Q&A]
Hello,

I'm using Micronaut version 2.5.12
I have the following issue - I have a Websocket Server application. I have also implemented a custom logic for closing a websocket session. In normal circumstances (like finishing the stream) it works correctly (it closes the session with specified code - taken from io.micronaut.websocket.CloseReason Enum), however, on a shutdown event it sends '1006: Abnormal closure' code instead of the one which I implemented in my shutdown hook. And the execution of close() method in fact doesn't close the session, it's only closed by Netty with above mentioned code (it is invoked from AbstractNettyWebsocketHandler, method handlerRemoved()).
Any idea how this can be resolved.
Please let me know if you need some more details.

Many thanks in advance for help.

**Comment by Rybenka** [comment]
@graemerocher , maybe you could advise something?

**Reply by graemerocher** [reply]
There have been some recent fixes in this area in 3.x, @yawkat can advise however we recommend you upgrade first

**Reply by yawkat** [reply]
Yes #6419 is related, so test with a version that has that fix first. If the issue persists, please report a bug with a test case
